docs/rfcs/internal-telemetry.md

```diff
@@ -0,0 +1,66 @@
+# Defining guidelines for internal telemetry
+
+## Overview
+
+The Collector supports generating internal telemetry
+that assists end users when operating the Collector. So
+far, much of the telemetry has been added to components as
+needed, without guidelines for component authors to provide
+a consistent experience to end users. The goal of this document
+is to:
+
+- describe the naming and attributes currently in use
+- define consistent units + naming for telemetry emitted by the Collector
+- define the process that should be used to configure new metrics
+
+## Out of scope
+
+This document is not intending to dictate when telemetry should be
+emitted by various Collector components. Considering the various types
+of components, this will be better discussed in a future document.
+
+This document is not intending to provide a comprehensive plan for how
+the Collector or the health of telemetry pipelines should be monitored. There
+is an [OpenTelemetry Enhancement Proposal](https://github.com/open-telemetry/oteps/pull/259) that has already started the process
+to provide this information. The information in this document will be updated once
+the OTEP lands to follow its recommendations.
+
+## Internal telemetry properties
+
+Telemetry produced by the Collector have the following properties:
+
+- metrics produced by Collector components use the prefix `otelcol_`. Until v0.106.0, this
+  was done via the Prometheus namespace, which made metrics inconsistent for users wishing
+  to emit metrics via another exporter. See https://github.com/open-telemetry/opentelemetry-collector/issues/9315
+  for more details
+- code is instrumented using the OpenTelemetry API and telemetry is produced via the OpenTelemetry Go SDK
+- instrumentation scope is defined via configuration in `metadata.yaml`
```

Comment:
Do we have any rules about the scope name?

Reply:
I don't know if rules is what i would call these, currently in contrib we have this pattern:

```
scope_name: otelcol/countconnector
scope_name: otelcol/datadog
scope_name: otelcol/exceptions
scope_name: otelcol/failover
scope_name: otelcol/grafanacloud
scope_name: otelcol/roundrobinconnector
scope_name: otelcol/routing
scope_name: otelcol/servicegraph
scope_name: otelcol/spanmetrics
scope_name: otelcol/sumconnector
scope_name: otelcol/alertmanager
scope_name: otelcol/alibabacloudlogservice
scope_name: otelcol/awscloudwatchlogs
scope_name: otelcol/awsemf
scope_name: otelcol/awskinesis
scope_name: otelcol/awss3
scope_name: otelcol/awsxray
scope_name: otelcol/azuredataexplorer
scope_name: otelcol/azuremonitor
scope_name: otelcol/carbon
scope_name: otelcol/cassandra
scope_name: otelcol/clickhouse
scope_name: otelcol/coralogix
scope_name: otelcol/datadog
scope_name: otelcol/dataset
scope_name: otelcol/elasticsearch
scope_name: otelcol/file
scope_name: otelcol/googlecloud
scope_name: otelcol/googlecloudpubsub
scope_name: otelcol/googlemanagedprometheus
scope_name: otelcol/honeycombmarker
scope_name: otelcol/influxdb
scope_name: otelcol/instana
scope_name: otelcol/kafka
scope_name: otelcol/kinetica
scope_name: otelcol/loadbalancing
scope_name: otelcol/logicmonitor
scope_name: otelcol/logzio
scope_name: otelcol/loki
scope_name: otelcol/mezmo
scope_name: otelcol/opencensus
scope_name: otelcol/opensearch
scope_name: otelcol/otelarrow
scope_name: otelcol/prometheus
scope_name: otelcol/prometheusremotewrite
scope_name: otelcol/pulsar
scope_name: otelcol/rabbitmq
scope_name: otelcol/sapm
scope_name: otelcol/sentry
scope_name: otelcol/signalfx
scope_name: otelcol/splunkhec
scope_name: otelcol/sumologic
scope_name: otelcol/syslog
scope_name: otelcol/tencentcloudlogservice
scope_name: otelcol/zipkin
scope_name: otelcol/ack
scope_name: otelcol/asapauth
scope_name: otelcol/awsproxy
scope_name: otelcol/basicauth
scope_name: otelcol/bearertokenauth
scope_name: otelcol/avrologencoding
scope_name: otelcol/jaegerencoding
scope_name: otelcol/jsonlogencoding
scope_name: otelcol/otlpencoding
scope_name: otelcol/textencoding
scope_name: otelcol/zipkinencoding
scope_name: otelcol/headerssetter
scope_name: otelcol/healthcheck
scope_name: otelcol/httpforwarder
scope_name: otelcol/jaegerremotesampling
scope_name: otelcol/oauth2clientauth
scope_name: otelcol/cfgardenobserver
scope_name: otelcol/dockerobserver
scope_name: otelcol/ecsobserver
scope_name: otelcol/ecstaskobserver
scope_name: otelcol/hostobserver
scope_name: otelcol/k8sobserver
scope_name: otelcol/oidcauth
scope_name: otelcol/opamp
scope_name: otelcol/pprof
scope_name: otelcol/remotetap
scope_name: otelcol/sigv4auth
scope_name: otelcol/solarwindsapmsettings
scope_name: otelcol/dbstorage
scope_name: otelcol/filestorage
scope_name: otelcol/sumologic
scope_name: otelcol/fileconsumer
scope_name: otelcol/attributes
scope_name: otelcol/cumulativetodelta
scope_name: otelcol/deltatocumulative
scope_name: otelcol/deltatorate
scope_name: otelcol/filter
scope_name: otelcol/groupbyattrs
scope_name: otelcol/groupbytrace
scope_name: otelcol/interval
scope_name: otelcol/k8sattributes
scope_name: otelcol/logstransform
scope_name: otelcol/metricsgeneration
scope_name: otelcol/metricstransform
scope_name: otelcol/probabilisticsampler
scope_name: otelcol/redaction
scope_name: otelcol/remotetap
scope_name: otelcol/resourcedetection
scope_name: otelcol/resource
scope_name: otelcol/routing
scope_name: otelcol/schema
scope_name: otelcol/span
scope_name: otelcol/sumologic
scope_name: otelcol/tailsampling
scope_name: otelcol/transform
scope_name: otelcol/activedirectorydsreceiver
scope_name: otelcol/aerospikereceiver
scope_name: otelcol/apachereceiver
scope_name: otelcol/apachesparkreceiver
scope_name: otelcol/awscloudwatchmetricsreceiver
scope_name: otelcol/awscloudwatchreceiver
scope_name: otelcol/awscontainerinsightreceiver
scope_name: otelcol/awsecscontainermetricsreceiver
scope_name: otelcol/awsfirehosereceiver
scope_name: otelcol/awsxrayreceiver
scope_name: otelcol/azureblobreceiver
scope_name: otelcol/azureeventhubreceiver
scope_name: otelcol/azuremonitorreceiver
scope_name: otelcol/bigipreceiver
scope_name: otelcol/carbonreceiver
scope_name: otelcol/chronyreceiver
scope_name: otelcol/cloudflarereceiver
scope_name: otelcol/cloudfoundryreceiver
scope_name: otelcol/collectdreceiver
scope_name: otelcol/couchdbreceiver
scope_name: otelcol/datadogreceiver
scope_name: otelcol/dockerstatsreceiver
scope_name: otelcol/elasticsearchreceiver
scope_name: otelcol/expvarreceiver
scope_name: otelcol/filelogreceiver
scope_name: otelcol/filestatsreceiver
scope_name: otelcol/flinkmetricsreceiver
scope_name: otelcol/fluentforwardreceiver
scope_name: otelcol/gitproviderreceiver
scope_name: otelcol/googlecloudpubsubreceiver
scope_name: otelcol/googlecloudspannerreceiver
scope_name: otelcol/haproxyreceiver
scope_name: otelcol/hostmetricsreceiver/cpu
scope_name: otelcol/hostmetricsreceiver/disk
scope_name: otelcol/hostmetricsreceiver/filesystem
scope_name: otelcol/hostmetricsreceiver/load
scope_name: otelcol/hostmetricsreceiver/memory
scope_name: otelcol/hostmetricsreceiver/network
scope_name: otelcol/hostmetricsreceiver/paging
scope_name: otelcol/hostmetricsreceiver/processes
scope_name: otelcol/hostmetricsreceiver/process
scope_name: otelcol/hostmetricsreceiver
scope_name: otelcol/httpcheckreceiver
scope_name: otelcol/iisreceiver
scope_name: otelcol/influxdbreceiver
scope_name: otelcol/jaegerreceiver
scope_name: otelcol/jmxreceiver
scope_name: otelcol/journaldreceiver
scope_name: otelcol/k8sclusterreceiver
scope_name: otelcol/k8seventsreceiver
scope_name: otelcol/k8sobjectsreceiver
scope_name: otelcol/kafkametricsreceiver
scope_name: otelcol/kafkareceiver
scope_name: otelcol/kubeletstatsreceiver
scope_name: otelcol/lokireceiver
scope_name: otelcol/memcachedreceiver
scope_name: otelcol/mongodbatlasreceiver
scope_name: otelcol/mongodbreceiver
scope_name: otelcol/mysqlreceiver
scope_name: otelcol/namedpipereceiver
scope_name: otelcol/nginxreceiver
scope_name: otelcol/nsxtreceiver
scope_name: otelcol/opencensusreceiver
scope_name: otelcol/oracledbreceiver
scope_name: otelcol/osqueryreceiver
scope_name: otelcol/otelarrowreceiver
scope_name: otelcol/otlpjsonfilereceiver
scope_name: otelcol/podmanreceiver
scope_name: otelcol/postgresqlreceiver
scope_name: otelcol/prometheusreceiver
scope_name: otelcol/pulsarreceiver
scope_name: otelcol/purefareceiver
scope_name: otelcol/purefbreceiver
scope_name: otelcol/rabbitmqreceiver
scope_name: otelcol/receivercreator
scope_name: otelcol/redisreceiver
scope_name: otelcol/riakreceiver
scope_name: otelcol/saphanareceiver
scope_name: otelcol/sapmreceiver
scope_name: otelcol/signalfxreceiver
scope_name: otelcol/simpleprometheusreceiver
scope_name: otelcol/skywalkingreceiver
scope_name: otelcol/snmpreceiver
scope_name: otelcol/snowflakereceiver
scope_name: otelcol/solacereceiver
scope_name: otelcol/splunkenterprisereceiver
scope_name: otelcol/splunkhecreceiver
scope_name: otelcol/sqlqueryreceiver
scope_name: otelcol/sqlserverreceiver
scope_name: otelcol/sshcheckreceiver
scope_name: otelcol/statsdreceiver
scope_name: otelcol/syslogreceiver
scope_name: otelcol/tcplogreceiver
scope_name: otelcol/udplogreceiver
scope_name: otelcol/vcenterreceiver
scope_name: otelcol/wavefrontreceiver
scope_name: otelcol/webhookeventreceiver
scope_name: otelcol/windowseventlogreceiver
scope_name: otelcol/windowsperfcountersreceiver
scope_name: otelcol/zipkinreceiver
scope_name: otelcol/zookeeperreceiver
```

and in core we use the package name:

```
        return settings.MeterProvider.Meter("go.opentelemetry.io/collector/internal/receiver/samplereceiver")
        return settings.MeterProvider.Meter("go.opentelemetry.io/collector/exporter/exporterhelper")
        return settings.MeterProvider.Meter("go.opentelemetry.io/collector/processor/batchprocessor")
        return settings.MeterProvider.Meter("go.opentelemetry.io/collector/processor/processorhelper")
        return settings.MeterProvider.Meter("go.opentelemetry.io/collector/receiver/receiverhelper")
        return settings.MeterProvider.Meter("go.opentelemetry.io/collector/receiver/scraperhelper")
        return settings.MeterProvider.Meter("go.opentelemetry.io/collector/service")
```

Reply:
@mx-psi issue https://github.com/open-telemetry/opentelemetry-collector/issues/9494 is opened about this